frontend/webpack.config.js

```diff
@@ -0,0 +1,24 @@
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const path = require('path');
+const APP_PATH = path.resolve(__dirname, 'src');
+
+module.exports = {
+  entry: APP_PATH,
+
+  output: {
+    filename: 'bundle.js',
+    path: path.resolve(__dirname, 'build'),
+  },
+
+  resolve: {
+    extensions: ['.ts', '.tsx', '.js', '.json']
+  },
+
+  module: {
+    rules: [{ test: /\.(ts|js)x?$/, loader: 'babel-loader', exclude: /node_modules/ }],
```

Comment:
I get the performance argument for excluding `node_modules` here, but philosophically I fall firmly in the camp of believing that it is the libraries' consumers' job to transpile to the target platform, since the libraries' authors can't know what that is. And of course some library authors agree with me, so this could leave us with untranspiled code in the bundle.

If we only target modern browsers, this is **probably** fine, and the speed boost is probably worth it, but **morally** I'm against it. x)

Reply:
Mhhh wait. So you are saying that some people might upload TS code to the npm registry? I mean, that sounds fine by me. But are you then saying that this then fails if webpack doesn't process the whole of `node_modules`? Like... what? I would assume that webpack checks what is imported in *our* code, then loads that module from `node_modules` and performs all transformations on it that are necessary (transpiling TS, minifiying, stripping unused code, ...). 

How does this webpack thingy even work? Like I'm already not happy with the whole state of affairs regarding webpack and JS tooling, please don't tell me it's worse than I thought :D

Reply:
I'm no Webpack expert, but here is my understanding:

Yes, Webpack looks at our imports and resolves them, i.e. it looks in certain places to find the correct files, and does transformations to them and then bundles everything together.

The details are described here: https://webpack.js.org/concepts/module-resolution/

This config snippet here does not say that all files with `(ts|js)x?` extension, except for those in `node_modules`, will be transformed. It says that when the resolution process described above finds a file with that extension, except when it is in `node_modules`, it should use `babel-loader` to transform the file before bundling it in.

That means that if we import something from an npm package that resolves to a `ts` file, it will not be compiled. This is probably fine, since you should probably not upload TS (directly) to npm, IMO. It is still a repository of JS-packages, after all. However, this also says that if we import something from `node_modules` that resolves to a `js`-file, we won't "babelify" it, i.e. transform all those cool new browser features to ES5 for older browsers. This is **probably** also mostly fine, since apparently many ppl babelify their code before pushing it to NPM, but it's not wrong to **not** do this, and some people don't, so we might end up shipping code that some older browsers can't run.

Reply:
I indeed don't like the way all this works, but this should be fixed by 5047a8b27e7834847c18f5a822c37b35caf378c4 now